dgraphtest/local_cluster.go

```diff
@@ -433,3 +437,44 @@ func (c *LocalCluster) AssignUids(num uint64) error {
 	}
 	return nil
 }
+
+// CompareCommits compare given commit with current cluster's commit and return
+// true if cluster's commit is ancestor of given commit else return false
+func CompareCommits(shaToBeCompare, baseSha string) (bool, error) {
+	repo, err := git.PlainOpen(repoDir)
+	if err != nil {
+		return false, errors.Wrap(err, "error while opening git repo")
+	}
+	hash, err := repo.ResolveRevision(plumbing.Revision(shaToBeCompare))
+	if err != nil {
+		return false, errors.Wrap(err, "error while getting commit hash")
+	}
+	commitToBeCompare, err := repo.CommitObject(*hash)
+	if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("error while getting commit object of hash [%v]", hash))
```

Comment:
use errors.Wrapf instead. Use small but clear error such as "error finding commit object [%v]"

avoid terms such as "while", "of has" (in this case). It doesn't be hash if you think about it, it could be a tag too